### DISCUSSION TITLE
How to associate keys with file types?

### DISCUSSION THREAD

**Discussion by habsfanongit** [LazyVim/LazyVim#1599] [General]
I noticed the latest lazy.vim (v10.6.0) added a feature (https://github.com/folke/lazy.nvim/pull/1078) to associate keys with file types using a new ft property but I am unable to use it using the following configuration where I am attempting to use F5 key to trigger RustDebuggables only with rust files.

```
return {
  -- Modify nvim-dap
  {
    "mfussenegger/nvim-dap",
    -- stylua: ignore
    keys = {
      { "<F5>", "<cmd>RustDebuggables<cr>" , desc = "Start Debugger", ft="rust" },
    },
  },
}

```
These configs are loaded from the plugin directory. 
I am  using the latest version of lazyvim and everything seems to be up to date.

Thanks

**Comment by dpetka2001** [comment]
I tried the following in my config
```lua
  {
    "toppair/peek.nvim",
    build = "deno task --quiet build:fast",
    keys = {
      {
        "<leader>pp",
        function()
          local peek = require("peek")
          if peek.is_open() then
            peek.close()
          else
            peek.open()
          end
        end,
        desc = "Peek (Markdown Preview)",
        ft = "markdown",
      },
    },
    opts = { theme = "dark" },
  },
```
and it correctly registers the key only for `markdown` files. The thing I couldn't figure out, is how to make `which-key` aware of the prefix name.

**Comment by habsfanongit** [comment]
@dpetka2001 , for some reason even your technique didn't work for me.
Here is my second attempt but this time I am attempting to set the key in  `simrat39/rust-tools.nvim` plugin config.
```
return {
  "simrat39/rust-tools.nvim",
  ft = "rust",

  opts = {
    showDisassembly = "never",
  },
  keys = {

    { "<F5>", "<cmd>RustDebuggables<cr>", desc = "Start Debugger", ft = "rust" },
  },
}
```
The key works in rust files but it's not ignored in other files. Hitting F5 while in any other file results in this error `E492: Not an editor command: RustDebuggables`

**Reply by folke** [reply]
did you update lazy.nvim??

**Reply by folke** [reply]
It's also better to add this keymap to the lsp keys section.  See https://www.lazyvim.org/configuration/keymaps

**Reply by dpetka2001** [reply]
I can confirm that something is up with the way `dap-core` is configured. I tried changing the default mapping
```lua
  {
    "mfussenegger/nvim-dap",
    keys = {
      -- stylua: ignore
      -- { "<F5>", function() require("dap").continue() end, desc = "Continue", ft = "python" },
      { "<leader>dc", function() require("dap").continue() end, desc = "Continue", ft = "python" },
    },
  },
```
just for experimenting and I noticed that `<leader>dc` is not mentioned in the `which-key` menu when in a lua file and is mentioned correctly when in a python file. However, in the Lua file, when I set a breakpoint with `<leader>db`, then if I press `<leader>d` to pop up the which-key menu for debugging, the `<leader>dc` key is mentioned in the menu(it seems it gets reinserted?). I don't really know how the new `ft` under `keys` section works, so maybe @folke would be more appropriate to give an answer about that.

**Reply by folke** [reply]
There was indeed something wrong there, where the mapping could be converted to a global one.

Just pushed a fix to lazy

**Comment by habsfanongit** [comment]
@folke, the latest update fixed it.
Thanks
